src/engine/threaded_engine.cc

```diff
@@ -415,6 +415,25 @@ void ThreadedEngine::WaitForAll() {
   finished_cv_.wait(lock, [this]() {
       return pending_.load() == 0 || kill_.load();
     });
+  std::exception_ptr tmp;
+  if (!global_exception_refs_.empty()) {
+    // iterate through all exception refs
+    for (auto itr = global_exception_refs_.begin();
```

Comment:
shall we use range for with const reference? is much less noisy.

Reply:
changed

Reply:
Thanks for the review!